pyEDAA/UCIS/Cobertura.py

```diff
@@ -49,43 +49,52 @@
 class Class:
 	"""Represents a code element in the Cobertura coverage data model (Java-focused)."""
 
-	source_file: str
+	name: str
+	sourceFile: str
 	lines: Dict[int, int]
-	lines_valid: int
-	lines_covered: int
+	linesValid: int
+	linesCovered: int
 
-	def __init__(self, source_file: str):
-		self.source_file = source_file
+	def __init__(self, name: str, sourceFile: str):
+		self.name = name
+		self.sourceFile = sourceFile
 		self.lines = {}
-		self.lines_valid = 0
-		self.lines_covered = 0
+		self.linesValid = 0
+		self.linesCovered = 0
 
-	def add_statement(self, line: int, count: int) -> None:
-		assert line not in self.lines.keys()
-		self.lines[line] = count
-		self.lines_valid += 1
-		if count:
-			self.lines_covered += 1
-
-	def get_xml_node(self) -> etree.Element:
-		class_node = etree.Element("class")
-		class_node.attrib["name"] = self.source_file
-		class_node.attrib["filename"] = self.source_file
-		class_node.attrib["complexity"] = "0"
-		class_node.attrib["branch-rate"] = "0"
-		class_node.attrib["line-rate"] = str(self.lines_covered / self.lines_valid)
-		class_node.append(etree.Element("methods"))
-		lines_node = etree.SubElement(class_node, "lines")
+	def addStatement(self, line: int, hits: int) -> None:
+		assert line not in self.lines.keys(), "Duplicated line number"
+
+		self.lines[line] = hits
+		self.linesValid += 1
+
+		if hits:
+			self.linesCovered += 1
+
+	def getXmlNode(self) -> etree.Element:
+		classNode = etree.Element("class")
+		classNode.attrib["name"] = self.sourceFile
+		classNode.attrib["filename"] = self.sourceFile
+		classNode.attrib["complexity"] = "0"
+		classNode.attrib["branch-rate"] = "0"
+
+		try:
+			classNode.attrib["line-rate"] = str(self.linesCovered / self.linesValid)
+		except ZeroDivisionError:
+			classNode.attrib["line-rate"] = "1"
```

Comment:
Why is this an integer of 1 in case of an error?
The division calculates a float, right?

Reply:
No valid lines means that source file exist but contains no code lines or all lines are exluded from coverage.

Coverage.py also can export python coverage to cobertura format and uses "1" and "0" in such cases:
```
<package name="." line-rate="1" branch-rate="0" complexity="0">
	<classes>
		<class name="main.py" filename="main.py" complexity="0" line-rate="1" branch-rate="0">
			<methods/>
			<lines/>
		</class>
	</classes>
</package>
```

In https://github.com/edaa-org/pyEDAA.UCIS/pull/7/commits/a3989380939cc40dad5c58f093cfe2d9e3aa499f I used slightly different conversion to string:
`packageNode.attrib["line-rate"] = f"{rate:.16g}"`

It will be now more consistent and calculated float values 0.0 and 1.0 will be put in xml file as "0" and "1"